share/p2p/shrexnd/client.go

```diff
@@ -178,15 +178,17 @@ func (c *Client) setStreamDeadlines(ctx context.Context, stream network.Stream)
 		return
 	}
 
-	if c.params.readTimeout != 0 {
-		err := stream.SetReadDeadline(time.Now().Add(c.params.readTimeout))
+	// client read deadline is the same as server write deadline
+	if c.params.serverWriteTimeout != 0 {
+		err := stream.SetReadDeadline(time.Now().Add(c.params.serverWriteTimeout))
```

Comment:
Why do we set write deadline on reading on the client side? 

Also, if we want to use write timeout as global timeout for both read and write ops we can use SetDeadline, but before would be nice to understand why we want a global one

Reply:
We expect it reasonable to write request in 5sec and if it didn't happen we could try another peer faster that whole request timeout happens. Full request timeout in general is 1 min (there are cases when it could be different).

Reply:
> We expect it reasonable to write request in 5sec and if it didn't happen we could try another peer faster that whole request timeout happens. 

Yeah, so why it's now `serverWriteTimeout` which is 1 min, instead of `serverReadTimeout` which is 5 secs?

Reply:
Because it client side timeouts (client.go). Client read timeout will either timeout itself or reach server write timeout. So for simplicity for now client read timeout == server write timeout.